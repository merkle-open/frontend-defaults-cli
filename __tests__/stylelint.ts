import { defaultIt } from './shared';

describe('stylelint', async () => {
	it('default', async () => await defaultIt('tmp-stylelint', '--stylelint --force --noInstall'));
	it('with-githooks', async () =>
		await defaultIt('tmp-stylelint-with-githooks', '--stylelint --githooks --force --noInstall'));
});
