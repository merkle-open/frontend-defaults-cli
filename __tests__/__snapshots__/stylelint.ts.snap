// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylelint default 1`] = `
"{
  \\"options\\": {
    \\"ts\\": false,
    \\"tslint\\": false,
    \\"es\\": false,
    \\"eslint\\": false,
    \\"editorconfig\\": false,
    \\"prettier\\": false,
    \\"stylelint\\": true,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": false,
    \\"commitlint\\": false,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"lint\\": \\"npm-run-all lint:*\\",
          \\"lint:css\\": \\"stylelint src/**/*.*ss\\"
        },
        \\"devDependencies\\": {
          \\"@namics/stylelint-config\\": \\"0.2.1\\",
          \\"npm-run-all\\": \\"4.1.3\\",
          \\"stylelint\\": \\"9.5.0\\"
        }
      }
    },
    \\"stylelint.config.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = require('@namics/stylelint-config');\\\\n\\"
    },
    \\".stylelintignore\\": {
      \\"override\\": false,
      \\"data\\": \\"node_modules/\\\\nsrc/shared/base/reset/css/yui/reset.css\\"
    }
  }
}"
`;

exports[`stylelint default 2`] = `
"node_modules/
src/shared/base/reset/css/yui/reset.css"
`;

exports[`stylelint default 3`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\",
    \\"lint:css\\": \\"stylelint src/**/*.*ss\\"
  },
  \\"devDependencies\\": {
    \\"@namics/stylelint-config\\": \\"0.2.1\\",
    \\"npm-run-all\\": \\"4.1.3\\",
    \\"stylelint\\": \\"9.5.0\\"
  }
}"
`;

exports[`stylelint default 4`] = `
"module.exports = require('@namics/stylelint-config');
"
`;

exports[`stylelint with-githooks 1`] = `
"{
  \\"options\\": {
    \\"ts\\": false,
    \\"tslint\\": false,
    \\"es\\": false,
    \\"eslint\\": false,
    \\"editorconfig\\": false,
    \\"prettier\\": false,
    \\"stylelint\\": true,
    \\"gitignore\\": false,
    \\"npmrc\\": false,
    \\"readme\\": false,
    \\"githooks\\": true,
    \\"commitlint\\": false,
    \\"nodeVersion\\": false,
    \\"webpack\\": false,
    \\"install\\": false,
    \\"force\\": true,
    \\"dryRun\\": false,
    \\"mode\\": \\"api\\"
  },
  \\"mergedFiles\\": {
    \\"package.json\\": {
      \\"override\\": false,
      \\"data\\": {
        \\"scripts\\": {
          \\"lint\\": \\"npm-run-all lint:*\\",
          \\"lint:css\\": \\"stylelint src/**/*.*ss\\"
        },
        \\"husky\\": {
          \\"hooks\\": {
            \\"pre-commit\\": \\"lint-staged\\"
          }
        },
        \\"lint-staged\\": {
          \\"linters\\": {
            \\"src/**/*.*ss\\": [
              \\"stylelint\\"
            ]
          }
        },
        \\"devDependencies\\": {
          \\"@namics/stylelint-config\\": \\"0.2.1\\",
          \\"husky\\": \\"1.0.1\\",
          \\"lint-staged\\": \\"7.3.0\\",
          \\"npm-run-all\\": \\"4.1.3\\",
          \\"stylelint\\": \\"9.5.0\\"
        }
      }
    },
    \\".huskyrc\\": {
      \\"override\\": false,
      \\"data\\": \\"# ~/.huskyrc\\\\n# Note: it will run in /bin/sh shell\\\\nif command -v nodenv >/dev/null 2>&1; then\\\\n  eval \\\\\\"$(nodenv init -)\\\\\\"\\\\nfi\\"
    },
    \\"stylelint.config.js\\": {
      \\"override\\": false,
      \\"data\\": \\"module.exports = require('@namics/stylelint-config');\\\\n\\"
    },
    \\".stylelintignore\\": {
      \\"override\\": false,
      \\"data\\": \\"node_modules/\\\\nsrc/shared/base/reset/css/yui/reset.css\\"
    }
  }
}"
`;

exports[`stylelint with-githooks 2`] = `
"# ~/.huskyrc
# Note: it will run in /bin/sh shell
if command -v nodenv >/dev/null 2>&1; then
  eval \\"$(nodenv init -)\\"
fi"
`;

exports[`stylelint with-githooks 3`] = `
"node_modules/
src/shared/base/reset/css/yui/reset.css"
`;

exports[`stylelint with-githooks 4`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"npm-run-all lint:*\\",
    \\"lint:css\\": \\"stylelint src/**/*.*ss\\"
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"pre-commit\\": \\"lint-staged\\"
    }
  },
  \\"lint-staged\\": {
    \\"linters\\": {
      \\"src/**/*.*ss\\": [
        \\"stylelint\\"
      ]
    }
  },
  \\"devDependencies\\": {
    \\"@namics/stylelint-config\\": \\"0.2.1\\",
    \\"husky\\": \\"1.0.1\\",
    \\"lint-staged\\": \\"7.3.0\\",
    \\"npm-run-all\\": \\"4.1.3\\",
    \\"stylelint\\": \\"9.5.0\\"
  }
}"
`;

exports[`stylelint with-githooks 5`] = `
"module.exports = require('@namics/stylelint-config');
"
`;
